src/tool.cpp

```diff
@@ -234,6 +234,12 @@ DigParams getDigParams(const ItemGroupList &groups,
 	return DigParams(result_diggable, result_time, wear_i, result_main_group);
 }
 
+DigParams getDigParams(const ItemGroupList &groups,
+		const ToolCapabilities *tp)
+{
+	return getDigParams(groups, tp, 1000000);
```

Comment:
- Why are you hardcoding `time_from_last_punch`? This completely destroys the purpose of basing the tool's `wear` upon this value.
- Why are you overloadng this function just to pass an extra hardcoded parameter?

Reply:
aha, i'm afraid you greatly overestimate my own understanding of the code - i was copying it directly from [the commit prior to removal](https://github.com/minetest/minetest/commit/345e1041a204ca9d40a25b71ae239ce3275f0755?diff=unified#diff-b2e9183274c47c49c500e499eacd627eL235). i'll see if the original pr has it differently.

Reply:
seems that the original pr [did indeed have this as well]( https://github.com/minetest/minetest/pull/6804/files#diff-b2e9183274c47c49c500e499eacd627eR234).

Reply:
Hm, that's weird. I assume the intention was to make this parameter optional.

Unless there's a reason I don't see, overloading a function just to specify the default value is... bad. A way more simpler, elegant, and correct approach is to directly specify a default value in the declaration of the main function.